magni_description/urdf/magni.urdf.xacro

```diff
@@ -181,29 +181,32 @@
     <!-- ahead    Directly facing forward with no tilt        -->
     <xacro:if value="${raspicam_mount == 'forward'}">
       <xacro:raspi_camera name="raspicam" connected_to="base_link">
-        <origin xyz="0.050 0.085 0.135" rpy="1.15191731 0 1.5707"/>
+        <origin xyz="0.050 0.085 0.135" rpy="0 ${-pi/2 - 1.15191731} ${pi}"/>
       </xacro:raspi_camera>
     </xacro:if>
     <xacro:if value="${raspicam_mount == 'upward'}">
       <xacro:raspi_camera name="raspicam" connected_to="base_link">
-        <origin xyz="0.020 0.115 0.155" rpy="0 0 ${pi}"/>
+        <origin xyz="0.020 0.115 0.155" rpy="0 ${-pi/2} ${-pi/2}"/>
       </xacro:raspi_camera>
     </xacro:if>
     <xacro:if value="${raspicam_mount == 'downward'}">
       <xacro:raspi_camera name="raspicam" connected_to="base_link">
-        <origin xyz="0.1 0.090 0.145" rpy="3.14159 -1.25 0.0"/>
+        <origin xyz="0.1 0.090 0.145" rpy="${-pi/2} ${pi/2 - 1.25} 0.0"/>

```

Comment:
When i run gazebo with "downward" camera is 90degrees into right: 
![image](https://user-images.githubusercontent.com/8426792/112463718-e00a1280-8d62-11eb-8da7-b68e6f9dd4da.png)

Probably it should be with the connector downward

Reply:
Well actually that might be exactly right, that's how the real life bracket we made positioned it. The reasoning was that we'd get a larger field of view vertically which would help with seeing markers far away.

Unfortunately the lack of seeing markers left and right is far worse than seeing those further forward since the detected poses will be garbage anyway. 

I'm not sure if that bracket will actually be used for anything at this point, but we should get a new one that doesn't rotate the camera and the implement a description pose for that.

Reply:
oh okay, i guess then for now these poses are good

Reply:
@

> When i run gazebo with "downward" camera is 90degrees into right:
> ![image](https://user-images.githubusercontent.com/8426792/112463718-e00a1280-8d62-11eb-8da7-b68e6f9dd4da.png)
> 
> Probably it should be with the connector downward

I have checked original camera positions (see the pictures below). And it was same as on your pictures. 
Are we leaving camera as it is or it should be rotated?

![Screenshot at ahead](https://user-images.githubusercontent.com/3756433/112657811-728ddd00-8e5b-11eb-96d7-12769cbea03f.png)
![Screenshot Down](https://user-images.githubusercontent.com/3756433/112657907-8a656100-8e5b-11eb-9628-f6cbce7e4c82.png)

Reply:
Yeah if the old ones are in side positions, then its cool with me, we can merge like it is

Reply:
No actually the "ahead" pose is wrong, that one should have the camera facing normally forward. Only the "downward" one is rotated 90 deg.

Reply:
@MoffKalast are you saying that from memory or do you have any source of info somewhere?
@Polonium19 I guess then just correct the "ahead" pose an then you can merge

Reply:
I'm pretty certain on the downward one, but not sure about ahead, I'm going off what I recall working on in TP, we should probably ask someone who's actually used it.

Reply:
Checked on a physical robot, "ahead" camera is not rotated, the cable connector is turned down.